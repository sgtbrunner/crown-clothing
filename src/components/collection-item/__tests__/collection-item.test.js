import React from 'react';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import CollectionItem from '../collection-item.component';

const mockStore = configureMockStore();

const props = {
  item: {
    price: 10,
    name: 'Item',
    imageUrl: 'https://www.myimage.com/item',
  },
  addItem: jest.fn(),
};

const mountComponent = () =>
  render(
    <Provider store={mockStore({})}>
      <CollectionItem {...props} />
    </Provider>
  );

describe('CollectionItem component', () => {
  const { name, price } = props.item;

  it('should match snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render collection item', () => {
    mountComponent();

    const collectionItem = screen.getByTestId('collection-item');
    expect(collectionItem).toBeInTheDocument();
  });

  it('should render collection item image', () => {
    mountComponent();

    const image = screen.getByRole('img');
    expect(image).toBeInTheDocument();
  });

  it('should render collection item text', () => {
    mountComponent();

    const itemText = screen.getByText(name);
    expect(itemText).toBeInTheDocument();
  });

  it('should render collection item price', () => {
    mountComponent();

    const itemPrice = screen.getByText(`$${price}`);
    expect(itemPrice).toBeInTheDocument();
  });

  it('should render "Add to cart" button', () => {
    mountComponent();

    const addButton = screen.getByText('Add to cart');
    expect(addButton).toBeInTheDocument();
  });
});
