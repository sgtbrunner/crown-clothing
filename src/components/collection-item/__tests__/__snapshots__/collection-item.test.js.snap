// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should match snapshot 1`] = `
<div>
  <div
    class="sc-gtsrHT Mxfvc"
  >
    <div
      class="sc-hKFxyN etKAyI"
      role="img"
    />
    <div
      class="sc-eCApnc flhIsE"
    >
      <span
        class="sc-jSFjdj duvpOf"
      >
        Item
      </span>
      <span>
        $10
      </span>
    </div>
    <button
      class="sc-bdnxRM pMdtA sc-dlnjwi kdkiUs"
    >
      Add to cart
    </button>
  </div>
</div>
`;
